### PR TITLE
Config server example

### PR DESCRIPTION
**Pull request by bmaidics**

### CODE REVIEW

File: http.filesystem.config.server/change_config.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -ex
+
+# change config file
+kubectl --namespace zilla-config-server create configmap configmap-served --from-file zilla.yaml -o yaml --dry-run=client | kubectl apply -f - > /dev/null 2>&1
+
+while [[ "$(curl -s -o -d 'Hello, World' -H 'Content-Type: text/plain' -X 'POST' /dev/null -w ''%{http_code}'' localhost:8080/echo_changed | sed 's/^0*//')" != "200" ]] > /dev/null 2>&1 ; do sleep 10 ; done
```
Comment (jfallows) [inline http.filesystem.config.server/change_config.sh:7]: Suggest using `curl -f ...` which has non-zero exit code on failure instead of `-w ''%{http_code}''` and parsing `stdout` for `200`.

Also, shouldn't `localhost:8080/echo_changed` be `http://localhost:8080/echo_changed`?

File: http.filesystem.config.server/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
++ sleep 10
++ sleep 10
++ sleep 10
++ sleep 10
+```
+Note: Kubernetes will not apply the changed configmap immediately, only after ~60 seconds.
+
+### Verify behavior of the reconfigured Zilla HTTP echo server
+
+### Verify the `/echo` path is not echoing anymore
```
Comment (jfallows) [inline http.filesystem.config.server/README.md:132]: Suggest rewording to `... path is no longer working`.

File: http.filesystem.config.server/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
++ kubectl --namespace zilla-config-server create configmap configmap-served --from-file zilla.yaml -o yaml --dry-run=client
++ kubectl apply -f -
++ sleep 10
++ sleep 10
++ sleep 10
++ sleep 10
++ sleep 10
++ sleep 10
+```
+Note: Kubernetes will not apply the changed configmap immediately, only after ~60 seconds.
```
Comment (jfallows) [inline http.filesystem.config.server/README.md:128]: Why does it take ~60secs to apply the change? I'd expect it to be fairly instantaneous.

File: http.filesystem.config.server/chart/templates/service-zilla-config.yaml [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+      containers:
+        - name: zilla-config
+          # TODO: change to latest when it gets merged
+          image: "ghcr.io/aklivity/zilla:develop-SNAPSHOT"
+          args: ["start", "-v", "-e"]
+          env:
+            - name: KEYSTORE_PASSWORD
+              value: generated
+            - name: JAVA_TOOL_OPTIONS
+              value: '-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005'
```
Comment (jfallows) [inline http.filesystem.config.server/chart/templates/service-zilla-config.yaml:26]: TODO: remove

File: http.filesystem.config.server/chart/templates/service-zilla-config.yaml [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+          ports:
+            - containerPort: 8081
+            - containerPort: 5005
```
Comment (jfallows) [inline http.filesystem.config.server/chart/templates/service-zilla-config.yaml:29]: TODO: remove

File: http.filesystem.config.server/chart/templates/service-zilla-config.yaml [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+metadata:
+  name: zilla
+data:
+  zilla.yaml: |-
+{{ .Files.Get "files/zilla.yaml" | indent 4 }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: configmap-served
```
Comment (jfallows) [inline http.filesystem.config.server/chart/templates/service-zilla-config.yaml:75]: Suggest renaming to `zilla-served` for parity with `zilla`, much like `tls` vs `tls-served` pattern below.

File: http.filesystem.config.server/chart/templates/service-zilla-config.yaml [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+spec:
+  selector:
+    app.kubernetes.io/instance: zilla-config
+  ports:
+    - name: http
+      port: 8081
+    - name: https
+      port: 9091
+    - name: debug
+      port: 5005
```
Comment (jfallows) [inline http.filesystem.config.server/chart/templates/service-zilla-config.yaml:111]: TODO: remove

File: http.filesystem.config.server/chart/templates/service-zilla-http.yaml [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+      containers:
+        - name: zilla-http
+          # TODO: change to latest when it gets merged
+          image: "ghcr.io/aklivity/zilla:develop-SNAPSHOT"
+          args: ["start", "-v", "-e", "-p", "/etc/zilla/zilla.properties"]
+          env:
+            - name: KEYSTORE_PASSWORD
+              value: generated
+            - name: JAVA_TOOL_OPTIONS
+              value: '-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005 -Djdk.internal.httpclient.debug=true'
```
Comment (jfallows) [inline http.filesystem.config.server/chart/templates/service-zilla-http.yaml:30]: TODO: remove

File: http.filesystem.config.server/chart/templates/service-zilla-http.yaml [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+          ports:
+            - containerPort: 8080
+            - containerPort: 8088
+            - containerPort: 5005
```
Comment (jfallows) [inline http.filesystem.config.server/chart/templates/service-zilla-http.yaml:34]: TODO: remove

File: http.filesystem.config.server/chart/templates/service-zilla-http.yaml [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+spec:
+  selector:
+    app.kubernetes.io/instance: zilla-http
+  ports:
+    - name: http-before
+      port: 8080
+    - name: http-after
+      port: 8088
+    - name: debug
+      port: 5005
```
Comment (jfallows) [inline http.filesystem.config.server/chart/templates/service-zilla-http.yaml:73]: TODO: remove

**Review by jfallows — CHANGES_REQUESTED** [review]
Let's review the service and config map names to make sure it's really clear which is which when people look over this more complex example with 2 differently configured zillas.

File: http.filesystem.config.server/README.md (hunk 3 above)
Comment (bmaidics) [reply http.filesystem.config.server/README.md:128]: See the answer here: https://kubernetes.io/docs/concepts/configuration/configmap/#mounted-configmaps-are-updated-automatically
I'm trying to come up with a solution to decrease it.

Comment (bmaidics) [reply http.filesystem.config.server/README.md:128]: I googled a lot. It seems like we'd need to change the syncPeriod in the kubeletConfiguration. That is a kubernetes cluster specific config. I think we're not able to change that in Docker Kubernetes cluster (at least I couldn't find a way to do it).

**Review by jfallows — APPROVED** [review]
